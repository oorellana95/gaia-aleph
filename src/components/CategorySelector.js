import React from 'react'

function CategorySelector(props) {
    const [val, setValue] = React.useState(0)

    return (
        <>
            <div className="blog_right_sidebar">
                <aside className="single_sidebar_widget post_category_widget">
                    <h4 className="widget_title">Category</h4>
                    <ul className="list cat-list">
                        {props.itemAll && (
                            <li key="0" id="li0" className="d-flex" onClick={() => { setValue(0); props.filterFunction(0) }}>
                                <p className={(val === 0 ? 'active' : '')}>All categories</p>
                            </li>
                        )}
                        {(props.items || !props.itemAll) && props.items.map(item => {
                            return (
                                <li className="d-flex" type="button" key={item.id} id={"li" + item.id} onClick={() => { setValue(item.id); props.filterFunction(item.id) }}>
                                    <p className={(val === item.id ? 'active' : '')}>{item.name}</p>
                                </li>
                            )
                        })}
                    </ul>
                    {props.error && <p>{props.error}</p>}
                </aside>
            </div>
        </>
    );
}

export default CategorySelector;