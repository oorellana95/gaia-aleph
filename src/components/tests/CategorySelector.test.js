import React from 'react'
import CategorySelector from '../CategorySelector';
import { shallow } from 'enzyme';

describe('CategorySelector from unit test', () => {
    const items = [{id: 1, name: "Item1"}, {id: 2, name: "Item2"}, {id: 3, name: "Item3"}]

    test('should render without crashing', () => {
        const wrapper = shallow(<CategorySelector items={items} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should crash if there are no items, neither itemAll', () => {
        expect(() => {shallow(<CategorySelector />)}).toThrow();
    });

    test('should crash if items are not an array', () => {
        expect(() => {shallow(<CategorySelector items="Hola" />)}).toThrow();
        expect(() => {shallow(<CategorySelector items={4} />)}).toThrow();
    });
    
    test('should have length 1, only the itemAll', () => {
        const wrapper = shallow(<CategorySelector itemAll={true}/>);
        expect(wrapper.find('li')).toHaveLength(1)
    });

    test('should have length 3 (only items)', () => {
        const wrapper = shallow(<CategorySelector items={items}/>);
        expect(wrapper.find('li')).toHaveLength(3)
    });

    test('should have length 4 (items + itemAll)', () => {
        const wrapper = shallow(<CategorySelector items={items} itemAll={true}/>);
        expect(wrapper.find('li')).toHaveLength(4)
    });

    test('should run onClick with the new state', () => {
        const updateFilter = jest.fn();
        const wrapper = shallow(<CategorySelector items={items} filterFunction={updateFilter}/>);
        wrapper.find('#li1').simulate('click');
        expect(updateFilter).toHaveBeenCalledWith(1)
    });
});