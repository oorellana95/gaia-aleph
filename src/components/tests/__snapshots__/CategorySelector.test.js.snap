// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelector from unit test it should render without crashing 1`] = `ShallowWrapper {}`;

exports[`CategorySelector from unit test should render without crashing 1`] = `ShallowWrapper {}`;
